File: packages/python-plugin-sdk/gordon_plugin_sdk/mock_host.py
from typing import Dict, Any, List
from .plugin import GordonPlugin
from .types import PluginExecutionContext

class MockPluginHost:
    def __init__(self, mock_tables: Dict[str, List[Dict[str, Any]]] = None):
        self.mock_tables = mock_tables or {}
        self.logs: List[str] = []

    def create_context(self, plugin_id: str) -> PluginExecutionContext:
        def query(sql: str) -> Dict[str, Any]:
            self.logs.append(f"[QUERY]: {sql}")
            for table_name, rows in self.mock_tables.items():
                if table_name.lower() in sql.lower():
                    return {"rows": rows, "rowCount": len(rows)}
            return {"rows": [], "rowCount": 0}

        def log(msg: str):
            self.logs.append(f"[LOG]: {msg}")

        return PluginExecutionContext(plugin_id, query, log)

    def run_tool(self, plugin: GordonPlugin, tool_name: str, params: Dict[str, Any]) -> Any:
        ctx = self.create_context(plugin.plugin_id)
        return plugin.execute_tool(tool_name, params, ctx)

File: packages/node-plugin-sdk/src/mock_host.ts
import { GordonPlugin } from './plugin.js';
import { PluginExecutionContext } from './types.js';

export interface MockHostOptions {
  mockTables?: Record<string, any[]>;
  mockTools?: Record<string, (params: any) => Promise<any>>;
}

export class MockPluginHost {
  private options: MockHostOptions;
  public logs: string[] = [];

  constructor(options: MockHostOptions = {}) {
    this.options = options;
  }

  public createExecutionContext(pluginId: string): PluginExecutionContext {
    return {
      pluginId,
      queryTable: async (sql: string) => {
        this.logs.push(`[MOCK QUERY]: ${sql}`);
        // Simple mock table extraction
        const match = sql.match(/FROM\s+["']?([a-zA-Z0-9_]+)["']?/i);
        const tableName = match ? match[1] : 'default';
        const rows = this.options.mockTables?.[tableName] || [];
        return { rows, rowCount: rows.length };
      },
      invokeHostTool: async (toolName: string, params: Record<string, any>) => {
        this.logs.push(`[MOCK TOOL INVOKE]: ${toolName}`);
        const handler = this.options.mockTools?.[toolName];
        if (!handler) {
          return { status: 'mock_success', toolName, params };
        }
        return handler(params);
      },
      log: (message: string) => {
        this.logs.push(`[PLUGIN LOG]: ${message}`);
      },
    };
  }

  public async runTool(plugin: GordonPlugin, toolName: string, params: any): Promise<any> {
    const context = this.createExecutionContext(plugin.manifest.id);
    return plugin.executeTool(toolName, params, context);
  }
}
